### PR TITLE
MWPW-130971 - Centralized 404

### PR DESCRIPTION
**Pull request by auniverseaway**
**Features**
* Maintains legacy compatibility
* Defaults to local 404 using 404 - local meta property
* Allows for federated 404 content via libs (using 404 - feds)
* Ability to have any design

**Under the hood**
* Added `base` to config
* Fragment correctly respects default content via css.

Resolves: [MWPW-130971](https://jira.corp.adobe.com/browse/MWPW-130971)

**Milo URLs (Federated)**
https://main--milo--adobecom.hlx.page/does-not-exist?martech=off
https://central-404--milo--adobecom.hlx.page/does-not-exist?martech=off
https://central-404--milo--adobecom.hlx.page/de/does-not-exist?martech=off

**CC URLs (Federated)**
https://milo-central-404--cc--aishwaryamathuria.hlx.page/this-doesnt-exist?milolibs=central-404
(from local repo)

https://main--cc--adobecom.hlx.live/this-doesnt-exist?milolibs=central-404
https://main--cc--adobecom.hlx.live/de/this-doesnt-exist?milolibs=central-404

**Bacom URLs (Legacy)**
https://main--bacom--adobecom.hlx.live/this-doesnt-exist?milolibs=central-404
https://main--bacom--adobecom.hlx.live/de/this-doesnt-exist?milolibs=central-404 (still pulls English)

**Milo College URLs (Local)**
https://main--milo-college--adobecom.hlx.page/does-not-exist/?milolibs=central-404

### CODE REVIEW

File: libs/templates/404/404.js [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+import { getConfig, createTag, loadArea, getMetadata } from '../../utils/utils.js';
 import { sampleRUM } from '../../utils/samplerum.js';
 
-async function load404() {
-  const config = getConfig();
+const { base, contentRoot, locale } = getConfig();
+
+async function get404(path) {
+  const { prefix } = locale;
+  const href = path || `${base}${prefix}/fragments/404#_dnt`;
```
Comment (chrischrischris) [inline libs/templates/404/404.js:8]: To verify, `#_dnt` used because it's already pointing to the localized path?

**Review by chrischrischris — APPROVED** [review]

**Review by mokimo — APPROVED** [review]

File: libs/templates/404/404.js (hunk 1 above)
Comment (auniverseaway) [reply libs/templates/404/404.js:8]: Yes. And the localized path is a bit non-standard... being after libs.

**Comment by auniverseaway** [comment]
@salonijain3 @aishwaryamathuria I am signing off of this feature so i can work on bootcamp / milo academy material. If you want to ship this, it should just need unit tests and some E2E validations.

Handing ownership over to you to take it across the finish line.

**Comment by aishwaryamathuria** [comment]
@auniverseaway @chrischrischris @mokimo 
I have added following changes to the PR.. Could you please re-review:
1. Adding Unit test to 404 template
2. Adding export to init function
3. Adding a separate IIFE to call init

File: libs/templates/404/404.js [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+  body.classList.remove('404');
+  body.classList.add('legacy-404');
+  const main = body.querySelector('main');
   main.innerHTML = '';
-  let resp = await fetch(`${config.locale.contentRoot}/404.plain.html`);
+  let resp = await fetch(`${locale.contentRoot}/404.plain.html`);
   if (!resp || !resp.ok) {
-    const root = config.contentRoot || '';
+    const root = contentRoot || '';
     resp = await fetch(`${root}/404.plain.html`);
```
Comment (rgclayton) [inline libs/templates/404/404.js:26]: @aishwaryamathuria Can you address this check warning?

Comment (aishwaryamathuria) [reply libs/templates/404/404.js:27]: @rgclayton I have updated the unit test to cover teh if condition

**Comment by aishwaryamathuria** [comment]
CC federated 404 in local repo - https://milo-central-404--cc--aishwaryamathuria.hlx.page/this-doesnt-exist?milolibs=central-404

**Review by mokimo — APPROVED** [review]

**Comment by spadmasa** [comment]
QA has verified the PR changes on the below 3 flows and details are updated in MWPW-130971
1) If 404 has 2 meta tags set "template"="404" and "404"="feds" then the 404 page will be fetched from milo libs ie. The page published from <milo sharepoint>/libs/404.docx will be used for 404 template
2) If 404 has 2 meta tags set "template"="404" and "404"="local" then the 404 page will be fetched from local fragments folder ie. The page from <sharepoint>/<content root>/fragments/404.docx will be used for 404 template
3) If 404 has only "template"="404"  meta tag set then legacy 404 "these are uncharted waters" template will be used form localeContentRoot folder fallback en folder.

**Pull request merged**